### PR TITLE
[FIX] point_of_sale: correct refund amount on rounded orders

### PR DESCRIPTION
**Pull request by robinengels**
Current behavior:
When an order total is rounded, if you refund it from the backend. The refund amount is not correct, and you will not be able to close your pos session.

Steps to reproduce:
- Create a rounding of 0.05 HALF UP and activate it on the PoS
- Create a product with a price of 49.99
- Sell the product in the PoS
- Go back to the backend without closing the session
- Refund the order from the backend
- Try to close the session

opw-3574913
---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by robinengels** [comment]
put the right amount in the popup

File: addons/point_of_sale/wizard/pos_payment.py [hunk 1]
```diff
@@ -53,10 +54,11 @@ def check(self):
         currency = order.currency_id
 
         init_data = self.read()[0]
+        payment_method = self.env['pos.payment.method'].browse(init_data['payment_method_id'][0])
         if not float_is_zero(init_data['amount'], precision_rounding=currency.rounding):
             order.add_payment({
                 'pos_order_id': order.id,
-                'amount': order._get_rounded_amount(init_data['amount']),
+                'amount': order._get_rounded_amount(init_data['amount'], payment_method.is_cash_count),
```
Comment (caburj) [inline addons/point_of_sale/wizard/pos_payment.py:61]: It's not only for cash, right? We also allow payment rounding on other payment methods. So I guess the boolean to pass here should be based on the cash_rounding config and the type of payment method.

Comment (robinengels) [reply addons/point_of_sale/wizard/pos_payment.py]: `_get_rounded_amount` already check if the config use rounding, so I just made sure we round all methods if the `only_round_cash_method` is not activated.

**Comment by robinengels** [comment]
@caburj `_get_rounded_amount` already check if the config use rounding, so I just made sure we round all methods if the `only_round_cash_method` is not activated.

**Comment by robinengels** [comment]
@caburj

**Review by caburj — APPROVED** [review]
robodoo r+

**Pull request closed without merging**